### ISSUE TITLE
add option for constant or 0 selectivity below first knot in cubic spline

### ISSUE THREAD

**Issue by k-doering-NOAA** [nmfs-stock-synthesis/stock-synthesis#86]
Imported from redmine, Issue [#82789](https://vlab.ncep.noaa.gov/redmine/issues/82789)
 Opened by @iantaylor-NOAA on 2020-09-11
Status when imported: In Progress
 
I'm working with a group at IATTC on adding a new "empirical selectivity" diagnostic in r4ss (which is just the ratio of observed catch at age or length to expected numbers at age or length). This project will also include a tool to externally estimate cubic spline or double normal selectivity parameters to use as starting values in an SS model. 

In the process of developing the external spline estimator, a recurring challenge is the tendency for the spline selectivity to curve upwards to the left of the first knot, a pattern that I've seen before. A useful feature for such cases (if not by default) would be an option to have constant or 0 selectivity prior to the first knot similar to the constant selectivity specified after the final knot.

I think I could help implement this if there was agreement to add it and what the user input should look like.

**Comment by iantaylor-NOAA** [comment]
Thanks @Rick-Methot-NOAA for reminding me of this issue. Looking back at the VLab redmine page (https://vlab.noaa.gov/redmine/issues/82789), I see that the checklist and attached figures added in October 2020 didn't come through to github, so I'm adding those here now.

I would also like to test out the previously undocumented option in ADMB for a "natural cubic spline" where you don't have to specify the slope at the first and last knot as discussed in this ADMB issue: https://github.com/admb-project/admb/issues/180. That setup may serve a similar purpose to the changes proposed here, but I'm not sure about that.

Commit b8e5b7f077597b58f6ba43414c03bdc49c253c84 implemented this new feature for length-based cubic splines as shown in the attached figures.

Proposed next steps 

- [ ]  modify implementation if needed
- [ ]  document new option in User Manual
- [ ]  implement for age-based cubic spline selectivity as well
- [ ]  test it out a bit more
- [ ]  merge spline_issue_82789 branch into the main branch

I think a 3.30.19 release goal is reasonable.

![spline_details](https://user-images.githubusercontent.com/4992918/135665231-3a49151f-4150-4cd6-ad7e-a03409c1f1e4.png)
![spline_compare](https://user-images.githubusercontent.com/4992918/135665238-d9f95fcd-f93a-4575-ad6d-1271af6fde9a.png)

**Comment by iantaylor-NOAA** [comment]
I got this option working for both length and age spline selectivity in the spline_issue_82789 branch. The figure below shows contrived examples for length-based (first knot at 40cm) and age-based (first knot at age-4) spline selectivity where it makes a difference. In the age case, the selectivity at age-0 was the maximum in the absence of this option so it changes the scaling for all ages rather than impacting just the bins below the first knot.

This option doesn't change the input or output for anyone who isn't interested in using the feature (just adds spline codes 10, 11 and 12 on top of the existing 0, 1, and 2 options), so I think it would be safe to create a pull request and merge the branch.

The model files I used for testing are in [BigSkate_splines_option10age.zip](https://github.com/nmfs-stock-synthesis/stock-synthesis/files/7737586/BigSkate_splines_option10age.zip), where it is fleets 4 and 5 with length-based spline selectivity and 3 and 6 with age-based.

In the process of working on this, I realized that for the age-based option it is somewhat redundant with the option to specify a minimum age of selected fish as described at https://nmfs-stock-synthesis.github.io/ss-documentation/SS330_User_Manual.html#:~:text=As%20of%20SS,are%20not%20selected). However, given that the user might be exploring multiple options for the positions of the knots, I think this new option would be slightly easier to use than having to revise the selectivity pattern input for any change in knot position to change the minimum age of selection.
![image](https://user-images.githubusercontent.com/4992918/146604215-0ecf16cb-43e9-4394-b12e-daba2febada7.png)
